### PR TITLE
CV  UI - Add versions screen, and capability to search/read all tables

### PR DESCRIPTION
**Pull request by sambible**
Airgun Support for the Versions screen of CV UI - supports reading/searching all tables. 

Robottelo PR exercising this: https://github.com/SatelliteQE/robottelo/pull/13478

### CODE REVIEW

File: airgun/entities/contentview_new.py [hunk 1]
```diff
@@ -42,12 +43,23 @@ def publish(self, entity_name, values=None):
             view.fill(values)
         view.next.click()
         view.finish.click()
-        view.progressbar.wait_for_result(delay=0.01)
+        # view.progressbar.wait_for_result(delay=0.01)
```
Comment (omkarkhatavkar) [inline airgun/entities/contentview_new.py:46]: is that intentional? why not remove then

File: airgun/entities/contentview_new.py [hunk 2]
```diff
@@ -42,12 +43,23 @@ def publish(self, entity_name, values=None):
...
         view.wait_displayed()
         return view.versions.table.read()
 
+    def read_version_table(self, entity_name, version, version_table_name, search_param=None):
+        """Reads a specific table for a CV Version"""
+        view = self.navigate_to(self, 'Version', entity_name=entity_name, version=version)
+        self.browser.plugin.ensure_page_safe(timeout='5s')
+        view.wait_displayed()
+        # This allows dynamic access to the proper table
+        getattr(view, version_table_name).table.wait_displayed()
```
Comment (omkarkhatavkar) [inline airgun/entities/contentview_new.py:58]: `version_table_name ` isn't it is a tab rather table, From the https://github.com/SatelliteQE/robottelo/pull/13478/files#diff-53d63aa8641c98a3d50840beb53a9d237755a65ba059f8b7a9571226e793e4b7R94 I see this a tab. I would suggest use `tab_name` instead.

File: airgun/entities/contentview_new.py [hunk 3]
```diff
@@ -182,3 +193,18 @@ def step(self, *args, **kwargs):
...
+
+    VIEW = ContentViewVersionDetailsView
+
+    def prerequisite(self, *args, **kwargs):
+        return self.navigate_to(self.obj, 'Edit', **kwargs)
+
+    def step(self, *args, **kwargs):
+        version = kwargs.get('version')
+        self.parent.versions.search(version)
+        self.parent.versions.table.row(version=version)['Version'].widget.click()
```
Comment (vsedmik) [inline airgun/entities/contentview_new.py:210]: Looks like the navigation fails here to read the versions table. I wonder if it's due to ambiguity of locators between these checkboxes 
image
or something else.

Comment (sambible) [reply airgun/entities/contentview_new.py:210]: It's actually failing to find the Version tab when trying to navigate to the Versions table period, which is even more bizarre. I did find that the locator I was using for the tab was quite ambiguous though, where it was matching at least 3 things on the page, including a breadcrumb higher up. Changing that to be very specific, seeing if it helps.

Comment (vsedmik) [reply airgun/entities/contentview_new.py:210]: looks like it helped, PRT passed at https://github.com/SatelliteQE/robottelo/pull/14949

**Review by vsedmik — APPROVED** [review]

**Review by damoore044 — APPROVED** [review]
nice work!

File: airgun/views/contentview_new.py [hunk 4]
```diff
@@ -271,16 +273,88 @@ def before_fill(self, values=None):
...
+            column_widgets={
+                'Name': Text('.//a'),
+                'Type': Text('.//a'),
+                'Product': Text('.//a'),
+                'Content': Text('.//a'),
+            },
+        )
+
+    @View.nested
+    class rpmPackageGroups(Tab):
```
Comment (omkarkhatavkar) [inline airgun/views/contentview_new.py:320]: can you please remove this camel case  ?

**Review by omkarkhatavkar — APPROVED** [review]

**Comment by omkarkhatavkar** [comment]
@sambible Looks overall good, just a smaller non-blocking comment.

File: airgun/views/contentview_new.py (hunk 4 above)
Comment (vijaysawant) [reply airgun/views/contentview_new.py:320]: > can you please remove this camel case ?

the name of class has grapped from locator value (`TAB_LOCATOR`) 
are we supposed to follow any naming convension @omkarkhatavkar?

File: airgun/views/contentview_new.py [hunk 5]
```diff
@@ -271,16 +273,88 @@ def before_fill(self, values=None):
...
+        TAB_LOCATOR = ParametrizedLocator(
+            './/button[@data-ouia-component-id="cv-version-details-tabs-tab-rpmPackageGroups"]'
+        )
+        searchbox = PF4Search()
+        table = PatternflyTable(
+            component_id="content-view-version-details-rpm-package-groups-table",
```
Comment (vijaysawant) [inline airgun/views/contentview_new.py:326]:
```suggestion
            component_id='content-view-version-details-rpm-package-groups-table',
```

File: airgun/views/contentview_new.py [hunk 6]
```diff
@@ -271,16 +273,88 @@ def before_fill(self, values=None):
...
+        )
+
+    @View.nested
+    class errata(Tab):
+        TAB_LOCATOR = ParametrizedLocator(
+            './/button[@data-ouia-component-id="cv-version-details-tabs-tab-errata"]'
+        )
+        searchbox = PF4Search()
+        table = PatternflyTable(
+            component_id="content-view-version-details-errata-table",
```
Comment (vijaysawant) [inline airgun/views/contentview_new.py:340]:
```suggestion
            component_id='content-view-version-details-errata-table',
```

File: airgun/views/contentview_new.py [hunk 7]
```diff
@@ -271,16 +273,88 @@ def before_fill(self, values=None):
...
+        )
+
+    @View.nested
+    class rpmPackages(Tab):
+        TAB_LOCATOR = ParametrizedLocator(
+            './/button[@data-ouia-component-id="cv-version-details-tabs-tab-rpmPackages"]'
+        )
+        searchbox = PF4Search()
+        table = PatternflyTable(
+            component_id="content-view-version-details-rpm-packages-table",
```
Comment (vijaysawant) [inline airgun/views/contentview_new.py:310]:
```suggestion
            component_id='content-view-version-details-rpm-packages-table',
```

File: airgun/views/contentview_new.py [hunk 8]
```diff
@@ -271,16 +273,88 @@ def before_fill(self, values=None):
...
+    )
+
+    @View.nested
+    class repositories(Tab):
+        TAB_LOCATOR = ParametrizedLocator(
+            './/button[@data-ouia-component-id="cv-version-details-tabs-tab-repositories"]'
+        )
+        searchbox = PF4Search()
+        table = PatternflyTable(
+            component_id="content-view-version-details-repositories-table",
```
Comment (vijaysawant) [inline airgun/views/contentview_new.py:293]:
```suggestion
            component_id='content-view-version-details-repositories-table',
```

**Review by vijaysawant — CHANGES_REQUESTED** [review]

File: airgun/views/contentview_new.py (hunk 4 above)
Comment (omkarkhatavkar) [reply airgun/views/contentview_new.py:320]: We dont have the pattern convention that we define to use. It was not a non-blocking comment. it is fine

**Review by vijaysawant — APPROVED** [review]
Ack

**Pull request merged**